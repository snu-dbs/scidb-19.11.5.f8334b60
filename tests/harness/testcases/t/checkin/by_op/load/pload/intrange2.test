# Test integer input ranges using in-engine CSV loading.  See ticket #4273.

# It was interesting to compare intrange2.expected (shadow values)
# vs. intrange.expected.  In the shadow array, the intrange2 offset
# numbers were  actually more accurate, because they reflect the file
# offset into the original CSV file, *not* into the translated SciDB
# text format file!  Win!!!
# but shadow arrays are no longer supported.

--setup
--start-query-logging

--test

# Note the i16 attribute is nullable.
CREATE ARRAY a_flat <i8:int8,i16:int16 null,i32:int32,i64:int64,u8:uint8,u16:uint16,u32:uint32,u64:uint64,s:string>[i=0:*,1000,0]

# On input, 'csv:l' means strip CSV header line.
--shell --store --command "iquery -c $IQUERY_HOST -p $IQUERY_PORT -q "LOAD a_flat FROM '${TEST_DATA_DIR}/integer_range.csv' AS 'csv:l' ERRORS 99""
scan(a_flat)  -- Inserted by scan_doctor

--cleanup
remove(a_flat)
--shell --command "rm -f /tmp/intrange.csv"

--stop-query-logging
