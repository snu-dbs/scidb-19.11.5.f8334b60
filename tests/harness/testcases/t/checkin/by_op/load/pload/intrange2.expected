SCIDB QUERY : <CREATE ARRAY a_flat <i8:int8,i16:int16 null,i32:int32,i64:int64,u8:uint8,u16:uint16,u32:uint32,u64:uint64,s:string>[i=0:*,1000,0]>
Query was executed successfully

SCIDB QUERY : <iquery -c $IQUERY_HOST -p $IQUERY_PORT -q "LOAD a_flat FROM '${TEST_DATA_DIR}/integer_range.csv' AS 'csv:l' ERRORS 99">
Query was executed successfully

SCIDB QUERY : <scan(a_flat)  -- Inserted by scan_doctor>
{i} i8,i16,i32,i64,u8,u16,u32,u64,s
{0} null,0,0,0,0,0,0,0,'int8 overflow'
{1} 66,0,0,0,0,0,0,0,'int8 hex input'
{2} null,0,0,0,0,0,0,0,'int8 extra chars'
{3} 41,0,0,0,0,0,0,0,'int8 octal input is taken as decimal'
{4} null,0,0,0,0,0,0,0,'int8 underflow'
{5} 0,null,0,0,0,0,0,0,'int16 overflow'
{6} 0,-13570,0,0,0,0,0,0,'int16 hex input'
{7} 0,null,0,0,0,0,0,0,'int16 extra chars'
{8} 0,43,0,0,0,0,0,0,'int16 octal input is taken as decimal'
{9} 0,null,0,0,0,0,0,0,'int16 underflow'
{10} 0,0,null,0,0,0,0,0,'int32 overflow'
{11} 0,0,-559038737,0,0,0,0,0,'int32 hex input'
{12} 0,0,null,0,0,0,0,0,'int32 extra chars'
{13} 0,0,479,0,0,0,0,0,'int32 octal input is taken as decimal'
{14} 0,0,null,0,0,0,0,0,'int32 underflow'
{15} 0,0,0,null,0,0,0,0,'int64 overflow'
{16} 0,0,0,496704877035502,0,0,0,0,'int64 hex input'
{17} 0,0,0,null,0,0,0,0,'int64 extra chars'
{18} 0,0,0,123456789123456789,0,0,0,0,'int64 octal input is taken as decimal'
{19} 0,0,0,null,0,0,0,0,'int64 underflow'
{20} 0,0,0,0,null,0,0,0,'uint8 overflow'
{21} 0,0,0,0,254,0,0,0,'uint8 hex input'
{22} 0,0,0,0,254,0,0,0,'uint8 hex trailing spaces'
{23} 0,0,0,0,null,0,0,0,'uint8 octal input is taken as decimal: overflow'
{24} 0,0,0,0,0,null,0,0,'uint16 overflow'
{25} 0,0,0,0,0,48879,0,0,'uint16 hex input'
{26} 0,0,0,0,0,null,0,0,'uint16 hex extra chars'
{27} 0,0,0,0,0,727,0,0,'uint16 octal input is taken as decimal'
{28} 0,0,0,0,0,0,null,0,'uint32 overflow'
{29} 0,0,0,0,0,0,3134984174,0,'uint32 hex input'
{30} 0,0,0,0,0,0,null,0,'uint32 extra chars'
{31} 0,0,0,0,0,0,987654321,0,'uint32 octal input is taken as decimal'
{32} 0,0,0,0,0,0,0,null,'uint64 overflow'
{33} 0,0,0,0,0,0,0,13464654573481085694,'uint64 hex input'
{34} 0,0,0,0,0,0,0,null,'uint64 extra chars'
{35} 0,0,0,0,0,0,0,429496729600,'uint64 octal input is taken as decimal'
{36} null,0,0,0,0,0,0,0,'int8 using an empty string'
{37} 43,0,0,0,0,0,0,0,'int8 trailing spaces'
{38} 44,0,0,0,0,0,0,0,'int8 trailing tab'
{39} 0,null,0,0,0,0,0,0,'int16 nullable using an empty string'
{40} 0,0,0,0,0,0,45,0,'uint32 trailing spaces'
{41} 0,0,0,0,0,0,46,0,'uint32 trailing tabs'
{42} 0,0,0,0,0,0,0,0,'int16 nullable using a zero'
{43} 0,null,0,0,0,0,0,0,'int16 nullable using whitespace'
{44} 45,0,0,0,0,0,0,0,'int8 leading spaces'
{45} 0,0,0,0,254,0,0,0,'uint8 hex leading and trailing spaces'
{46} 0,null,0,0,0,0,0,0,'int16 nullable using Linear-TSV \\N null'
{47} 0,null,0,0,0,0,0,0,'int16 nullable using Linear-TSV \\N null and spaces'
{48} 0,?5,0,0,0,0,0,0,'int16 nullable using SciDB missing reason 5'
{49} 0,null,0,0,0,0,0,0,'int16 nullable using SciDB notation and spaces'
{50} 0,null,0,0,0,0,0,0,'int16 nullable using SciDB null'
{51} 0,null,0,0,0,0,0,0,'int16 nullable using SciDB null and spaces'

SCIDB QUERY : <remove(a_flat)>
Query was executed successfully

